manifests/server/pg_hba_rule.pp

```diff
@@ -16,7 +16,16 @@
   String $database,
   String $user,
   String $auth_method,
-  Optional[String] $address       = undef,
+  Optional[
+    Variant[
+      Stdlib::IP::Address::V4::CIDR,
+      Stdlib::IP::Address::V6::CIDR,
+      Stdlib::Fqdn,
+      Enum['all', 'samehost', 'samenet'],
+      # RegExp for a DNS domain - also starting with a single dot
+      Pattern[/^\.(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z0-9][a-zA-Z0-9-]{0,61}[a-zA-Z0-9]$/],
```

Comment:
Any chance that you could convert this to a local type, along the lines of https://github.com/puppetlabs/puppetlabs-apache/blob/main/types/modproxyprotocol.pp. the pattern at least.
As it is, this is kinda a lot

Reply:
Done.